### PR TITLE
#166, #187, #188

### LINKED ISSUE #166: [FEATURE] Automated Testing v2

### PR DESCRIPTION
**Pull request by cemersoz**
fixed
#166 
#187 
#188

### CODE REVIEW

File: server/imports/lab/checkLab.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = function(labfile){
   else{
     try{
       var  lab_obj = eval(labfile);
-      if(typeof lab_obj == undefined || typeof lab_obj == null){ //check for lab object
+      if(typeof lab_obj == "undefined" || typeof lab_obj == "null"){ //check for lab object
```
Comment (DerekTBrown) [inline server/imports/lab/checkLab.js]: typeof lab_obj === "undefined" || lab_obj === null

File: tests/example_data/courses/courses.js [hunk 2]
```diff
@@ -8,6 +8,10 @@ module.exports = [
 	course_name: "Great Practical Ideas for Computer Scientists",
 	course_number: "15-131",
 	instructors: [
+	  {
+            name: "Super User",
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: Indentation

File: tests/tuxlab.db.js [hunk 3]
```diff
@@ -7,7 +7,8 @@ var async = require('async');
 
 //example databases
 var labs_good_1 = require('./example_data/labs/labs.good.js');
```
Comment (DerekTBrown) [inline tests/tuxlab.db.js]: It would be nice to convert this into an array of good and bad labs, and have comments explaining why each lab should fail

Comment (cemersoz) [inline tests/tuxlab.db.js]: That is an array, and I have comments in the file

File: tests/example_data/courses/courses.js [hunk 4]
```diff
@@ -4,55 +4,63 @@
...
+        high-support way. We will discuss UNIX, LaTeX, debugging and many other essential tools.
+      `,
+      syllabus: "The syllabus is here."
+    },
+    permissions:{
+      meta : true,
+      content: "any",
+      enroll: "any"
+    },
+    hidden: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 5]
```diff
@@ -4,55 +4,63 @@
...
+    disabled: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 6]
```diff
@@ -62,97 +70,109 @@ module.exports = [
...
-  	featured: true,
-	labs: []
+      syllabus: 'The syllabus is here.'
+    },
+    permissions:{
+      meta: true,
+      content: "auth",
+      enroll: "none"
+    },
+    hidden: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 7]
```diff
@@ -62,97 +70,109 @@ module.exports = [
...
+    disabled: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 8]
```diff
@@ -62,97 +70,109 @@ module.exports = [
...
-  	featured: true,
-	labs: []
+      syllabus: 'The syllabus is here.'
+    },
+    permissions: {
+      meta: true,
+      content: "any",
+      enroll: "any"
+    },
+    hidden: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 9]
```diff
@@ -62,97 +70,109 @@ module.exports = [
...
+    disabled: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 10]
```diff
@@ -161,16 +181,16 @@ module.exports = [
...
-  	featured: true,
-	labs: []
+      syllabus: 'The syllabus is here.'
+    },
+    permissions:{
+      meta: true,
+      content: "any",
+      enroll: "any"
+    },
+      hidden: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 11]
```diff
@@ -161,16 +181,16 @@ module.exports = [
...
+      disabled: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 12]
```diff
@@ -62,97 +70,109 @@ module.exports = [
...
-    	featured: true,
-	labs: []
+        syllabus: 'The syllabus is here.'
+    },
+    permissions:{
+      meta: true,
+      content: "any",
+      enroll: "none"
+    },
+    hidden: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/example_data/courses/courses.js [hunk 13]
```diff
@@ -62,97 +70,109 @@ module.exports = [
...
+    disabled: false,
```
Comment (DerekTBrown) [inline tests/example_data/courses/courses.js]: This is gone

File: tests/tuxlab.db.js (hunk 3 above)
Comment (DerekTBrown) [inline tests/tuxlab.db.js]: I mean labs_good[0], lab_good[1] etc

**Pull request merged**